### ISSUE TITLE
Bad request with multiple (>2) expands using collections

### ISSUE THREAD

**Issue by lilyjw** [object/Simple.OData.Client#86]
Using Simple.OData.V4.Client 4.2.1, expanding nested collections "a/b" results in a successful query with ```$expand=a($expand=b)```, but expanding a deeper "a/b/c" results in a bad query with ```$expand=a($expand=b/c)``` - can you take a look at this?

For example, using a test from commit bb4e150 modified to hit a public service:
```
var client = new ODataClient("http://services.odata.org/V4/OData/OData.svc/");
var product = (await client
      .For("Products")
      .OrderBy("ID")
      .Expand("Category/Products/Category")
      .FindEntriesAsync()).Last();
```
Request:
 ```http://services.odata.org/V4/OData/OData.svc/Products?$expand=Categories($expand=Products/Categories)&$orderby=ID ```
Response:
```
HTTP/1.1 400 Bad Request
Message: Found a path traversing multiple navigation properties. Please rephrase the query such that each expand path contains only type segments and navigation properties
```

**Comment by object** [comment]
Thank you for the bug report. I will investigate the issue and get back to you.

**Comment by object** [comment]
Fixed in the version 4.3.0.
